src/control/content/components/LocationsActionBar.js

```diff
@@ -108,24 +108,56 @@ class LocationsActionBar extends React.Component {
             }).catch(() => undefined)
           );
         } else {
-          return {
-            id,
-            title:
-              typeof title === "number"
-                ? title.toString()
-                : title || "Untitled Location",
-            subtitle,
-            categories: selectedCategory,
-            address: {
-              name,
-              lat: parseFloat(address_lat),
-              lng: parseFloat(address_lng),
-            },
-            description,
-            image,
-            index: i + places.length,
-            indexForError: i
-          };
+        promises.push(
+            new Promise((resolve, reject) => {
+          let lat = address_lat;
+          let lng = address_lng;
+          let addressUrl = `https://maps.googleapis.com/maps/api/geocode/json?key=AIzaSyBOp1GltsWARlkHhF1H_cb6xtdR1pvNDAk&latlng=${lat},${lng}`
+              // get geodata from google api
+              fetch(addressUrl)
+                .then((response) => response.json())
+                .then((data) => {
+                  const name = data.results[0].formatted_address
+                  resolve({
+                    id,
+                    title:
+                      typeof title === "number"
+                        ? title.toString()
+                        : title || "Untitled Location",
+                    subtitle,
+                    categories: selectedCategory,
+                    address: {
+                      name,
+                      lat,
+                      lng,
+                    },
+                    description,
+                    image,
+                    index: i + places.length,
+                    indexForError: i
+                  });
+                });
+            }).catch(() => undefined)
```

Comment:
why do we need the catch if its not used?

Reply:
Removed the catch